components/p3a/brave_p3a_service.h

```diff
@@ -51,35 +53,57 @@ class BraveP3AService : public base::RefCountedThreadSafe<BraveP3AService>,
   // updates. Can't call it in constructor because of refcounted peculiarities.
   void InitCallbacks();
 
+  // Called by other components to add/remove dynamic metrics
+  // (metrics not included in the metric_names.h static list)
+  void RegisterDynamicMetric(const std::string& histogram_name,
+                             MetricLogType log_type,
+                             bool should_be_on_ui_thread = true);
```

Comment:
I would not expose this option on the public api, it should instead call `RegisterDynamicMetricImpl` internally and just leave `DCheckCurrentlyOnUIThread` as part of `RegisterDynamicMetric`